python/sparktk/frame/ops/categorical_summary.py

```diff
@@ -1,4 +1,181 @@
+    The returned value is a Map containing categorical summary for each specified column.
+
+    For each column, levels which satisfy the top k and/or threshold cutoffs are
+    displayed along with their frequency and percentage occurrence with respect to
+    the total rows in the dataset.
+
+    Performs level pruning first based on top k and then filters
+    out levels which satisfy the threshold criterion.
+
+    Missing data is reported when a column value is empty ("") or null.
+
+    All remaining data is grouped together in the Other category and its frequency
+    and percentage are reported as well.
+
+    User must specify the column name and can optionally specify top_k and/or threshold.
+
+    Examples
+    --------
+    Consider Frame *my_frame*, which contains the data
+
+    <hide>
+    >>> s = [("source",str),("target",str)]
+    >>> rows = [ ["entity","thing"], ["entity","physical_entity"],["entity","abstraction"],["physical_entity","entity"],["physical_entity","matter"],["physical_entity","process"],["physical_entity","thing"],["physical_entity","substance"],["physical_entity","object"],["physical_entity","causal_agent"],["abstraction","entity"],["abstraction","communication"],["abstraction","group"],["abstraction","otherworld"],["abstraction","psychological_feature"],["abstraction","attribute"],["abstraction","set"],["abstraction","measure"],["abstraction","relation"],["thing","physical_entity"],["thing","reservoir"],["thing","part"],["thing","subject"],["thing","necessity"],["thing","variable"],["thing","unit"],["thing","inessential"],["thing","body_of_water"]]
+    >>> my_frame = tc.to_frame(rows, s)
+    -etc-
+
+    </hide>
+    >>> my_frame.inspect()
+    [#]  source           target
+    =====================================
+    [0]  entity           thing
+    [1]  entity           physical_entity
+    [2]  entity           abstraction
+    [3]  physical_entity  entity
+    [4]  physical_entity  matter
+    [5]  physical_entity  process
+    [6]  physical_entity  thing
+    [7]  physical_entity  substance
+    [8]  physical_entity  object
+    [9]  physical_entity  causal_agent
+
+    >>> cm = my_frame.categorical_summary(tc.CategoricalSummaryInput('source', top_k=2))
+    <progress>
+
+    >>> cm
+    [column_name = source
+     levels      = [frequency  = 9
+     level      = thing
+     percentage = 0.321428571429, frequency  = 9
+     level      = abstraction
+     percentage = 0.321428571429, frequency  = 0
+     level      = Missing
+     percentage = 0.0, frequency  = 10
+     level      = Other
+     percentage = 0.357142857143]]
+
+    >>> cm = my_frame.categorical_summary(tc.CategoricalSummaryInput('source', threshold = 0.5))
+    <progress>
+
+    >>> cm
+    [column_name = source
+     levels      = [frequency  = 0
+     level      = Missing
+     percentage = 0.0, frequency  = 28
+     level      = Other
+     percentage = 1.0]]
```

Comment:
In this case the default __repr__ with nested prop objects is a little awkward.  I'd prefer a different repr here.